packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_services.dart

```diff
@@ -54,19 +70,17 @@ class DeepLinksServices {
       final List? failedChecks = domainResult[_failedChecksKey];
       if (failedChecks != null) {
         for (final Map<String, dynamic> failedCheck in failedChecks) {
-          switch (failedCheck[_checkNameKey]) {
-            case _existenceCheckKey:
-              domainErrors[domainName]!.add(DomainError.existence);
-            case _fingerPrintChecktKey:
-              domainErrors[domainName]!.add(DomainError.fingerprints);
+          final checkName = failedCheck[_checkNameKey];
+          if (checkNametoDomainError[checkName] != null) {
```

Comment:
nit: put `checkNametoDomainError[checkName]` into a local since it is used twice and since that would allow type promotion to a non-nullable type